### PR TITLE
feat(toplevel): adding in the toplevel v3 responses

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Add the top level response fields into the v3 rest response like:
* status
* complete
* error
* maxActions
* since

https://mozilla-hub.atlassian.net/browse/POCKET-9725

## Not Implemented

This does not implement `error` codes or `complete` fully. we will followup with those

### CODE REVIEW

File: servers/v3-proxy-api/src/graph/get/toRest.ts [hunk 1]
```diff
@@ -35,6 +38,30 @@ type SavedItemSimple =
...
+  unconfirmed_shares: [],
+};
+
+function statusResponse(
+  response:
+    | GetSavedItemsByOffsetCompleteQuery
+    | GetSavedItemsByOffsetSimpleQuery,
+): GetTopLevelDefaultResponse {
+  return {
+    status: response.user.savedItemsByOffset.totalCount > 0 ? 1 : 2,
```
Comment (bassrock) [inline servers/v3-proxy-api/src/graph/get/toRest.ts:58]: Need to fill this out..

File: servers/v3-proxy-api/src/test/fixtures/fixtures.ts [hunk 2]
```diff
@@ -103,7 +103,7 @@ type MockListItemObject = Omit<ListItemObject, 'item_id' | 'resolved_id'> & {
 /**
  * return REST v3 GET response for given Ids
  * //todo: need to include tags, images, videoes etc
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/test/fixtures/fixtures.ts:105]: I actually did this but missed this comment

both this todo and the todo underneath are now done thanks to our combined efforts 💪

File: servers/v3-proxy-api/src/graph/types.ts [hunk 3]
```diff
@@ -66,18 +66,39 @@ export type GetSearchResponseCompleteTotal = GetSearchResponseComplete & {
...
+ */
+export type GetSharesResponse = {
+  // empty data fields that used to have a use, but now are empty and Android will crash without.
+  recent_friends: [];
+  auto_complete_emails: [];
+  unconfirmed_shares: [];
+};
+
+export type GetTopLevelDefaultResponse = {
+  complete: number; // 0 if preg_match('/^[0-9]*$/', $since) && $since > 0, else 1
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/graph/types.ts:89]: aha, _that's_ why it didn't match `detailType=complete`

File: servers/v3-proxy-api/src/graph/get/toRest.ts [hunk 4]
```diff
@@ -35,6 +38,97 @@ type SavedItemSimple =
 type SavedItemComplete =
   GetSavedItemsByOffsetCompleteQuery['user']['savedItemsByOffset']['entries'][number];
 
+const staticV3ResponseDefaults: GetStaticResponse = {
```
Comment (kschelonka) [inline servers/v3-proxy-api/src/graph/get/toRest.ts:41]: If you don't mind, please put short docstrings on these functions

**Review by kschelonka — APPROVED** [review]
Request some docstrings but looks good otherwise (don't need to re-review)

File: servers/v3-proxy-api/src/graph/types.ts (hunk 3 above)
Comment (bassrock) [reply servers/v3-proxy-api/src/graph/types.ts:92]: yea i still gotta understand and come back to this one.

File: servers/v3-proxy-api/src/test/fixtures/fixtures.ts (hunk 2 above)
Comment (bassrock) [reply servers/v3-proxy-api/src/test/fixtures/fixtures.ts]: YAY!

**Pull request merged**